pkg/ebpf/tracee.go

```diff
@@ -802,6 +826,55 @@ func (t *Tracee) populateBPFMaps() error {
 	return nil
 }
 
+var syscallsToCheckX86 = []int{
+	0,   // read
+	1,   // write
+	2,   // open
+	3,   // close
+	41,  // socket
+	44,  // sendto
+	45,  // recvfrom
+	46,  // sendmsg
+	47,  // recvmsg
+	59,  // execve
+	62,  // kill
+	78,  // getdents
+	101, // ptrace
+	217, // getdents64
+	321, // bpf
+	322, // execveat
+	16,  // ioctl
+	257, // openat
+}
+
+var syscallsToCheckArm = []int{
+	63, // read
+	64, // write
+	// open (not present)
+	57,  // close
+	29,  // ioctl
+	198, // socket
+	221, // execve
+	129, // kill
+	// getdents (not present)
+	117, // ptrace
+	61,  // getdents64
+	280, // bpf
+	281, // execveat
+	56,  // openat
+	206, // sendto
+	207, // recvfrom
+	211, // sendmsg
+	212, // recvmsg
+}
+
+func getSyscallList(arch string) []int {
+	if strings.Compare(arch, "x86_64") == 0 {
+		return syscallsToCheckX86
+	}
+	return syscallsToCheckArm
+}
```

Comment:
instead of having such a function, you can put the above two lists in a file that is conditionaly compiled for a given architecture, and use the same variable name for the two tables. For example, you can simply put the tables under events_amd64.go and events_amd64.go.